Code/GraphMol/Kekulize.cpp

```diff
@@ -571,24 +597,48 @@ void KekulizeFragment(RWMol &mol, const boost::dynamic_bitset<> &atomsToUse,
     }
     const VECT_INT_VECT &allrings =
         allringsSSSR.empty() ? mol.getRingInfo()->atomRings() : allringsSSSR;
+    std::deque<INT_VECT> tmpRings;
+    auto containsNonDummy = [&atomsToUse, &dummyAts](const INT_VECT &ring) {
+      bool ringOk = false;
+      for (auto ai : ring) {
+        if (!atomsToUse[ai]) {
+          return false;
+        }
+        if (!dummyAts[ai]) {
+          ringOk = true;
+        }
+      }
+      return ringOk;
+    };
+    // we can't just copy the rings over: we're going to rearrange them so that
+    // we try to favor starting the traversal of any ring from an atom that is
+    // at the end of a wedged ring bond. This is part of our attempt to avoid
+    // assigning double bonds to bonds with wedging
+    for (const auto &ring : allrings) {
+      if (containsNonDummy(ring)) {
+        unsigned int startPos = 0;
+        bool hasWedge = false;
+        for (auto ri = 0u; ri < ring.size(); ++ri) {
+          if (wedgedAtoms[ring[ri]]) {
+            startPos = ri;
+            hasWedge = true;
+            break;
+          }
+        }
+        INT_VECT nring(ring.size());
+        for (auto ri = 0u; ri < ring.size(); ++ri) {
+          nring[ri] = ring[(ri + startPos) % ring.size()];
```

Comment:
I would use `at()` here so if the index is for whatever reason out of bounds we get an `out_of_range` exception rather than a segfault:
```suggestion
          nring[ri] = ring.at((ri + startPos) % ring.size());
```